spec/workers/sha_poller_worker_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe ShaPollerWorker, type: :worker do
+  before {
+    allow(AppEnv).to receive(:[]).with('HEROKU_SLUG_COMMIT') { nil }
+  }
+
```

Comment:
what does this do? why have you passed 'HEROKU_SLUG_COMMIT' as a parameter ?

Reply:
For the unit test, we are assuming AppEnv has the function `[]` with the variable name `'HEROKU_SLUG_COMMIT'`, as you would call `AppEnv['HEROKU_SLUG_COMMIT']`. It then expects a `nil` (null) value. We're basically telling the AppEnv object to return nil for the 'HEROKU_SLUG_COMMIT' variable.